docs/src/tutorials/MHMPC.md

```diff
@@ -0,0 +1,207 @@
+
+In order to connect nodes between the subgraphs, you must reference the `graph_master` due to the scope of the variables. `graph_master` has access to all variables from all subgraphs, after all subgraphs were added  to the master graph. The `@linkconstraint` structure here is the same as connecting between individual subgraphs; taking the expression from the last node in the subgraph then setting it equal to the first $x$ value of the next node.
+
+```julia
+# Adding linking constraints between nodes in individual SG's
+for (subgraph, cardinality, node) in zip(subgraphs_all, N_i, nodes_all)   
+    @linkconstraint(subgraph, [j = 1:(cardinality-1)], node[j+1][:x][1] == node[j][:Ax_Bu][1])
+    @linkconstraint(subgraph, [j = 1:(cardinality-1)], node[j+1][:x][2] == node[j][:Ax_Bu][2])
+end
+
+# Add subgraphs to master graph
+for subgraph in subgraphs_all
+    add_subgraph(graph_master, subgraph)
+end
+
+# Add linking constraint between subgraphs
+@linkconstraint(graph_master, [n = 1:(length(subgraphs_all)-1)], nodes_all[n+1][1][:x][1] == nodes_all[n][N_i[n]][:Ax_Bu][1])
+@linkconstraint(graph_master, [n = 1:(length(subgraphs_all)-1)], nodes_all[n+1][1][:x][2] == nodes_all[n][N_i[n]][:Ax_Bu][2])
+  
+```
+
+### 5. Setting objective and running optimization
+
+The final step is to set the objective and run the optimization. This step is written out in the mathematic formulation and is just written out. The code to run the optimization is straight forward as well. `set_to_node_objectives` is a newer function that is required for solving the optimization problem.
+
+The resulting output is the optimal values for $x, u$.
+
+```julia
+# Set objective
+@objective(
+    graph_master, 
+    Min, 
+    sum(node[:u]' * R * node[:u] for node in all_nodes(graph_master)) +
+    sum(node[:x]' * Q * node[:x] for node in all_nodes(graph_master)),
+)
+
+# Run optimization
+set_to_node_objectives(graph_master)
+set_optimizer(graph_master, Ipopt.Optimizer)
+set_optimizer_attribute(graph_master, "print_level", 0) # Silence the optimization output
+optimize!(graph_master)
+```
+
+### 6. Run the program
+
+In order to run this optimization simulation, we must take all of this work and run it in real time. In context of the spring damper system, we want to apply an optimal force $u$ to dampen the system to position $0$, and $0$ velocity. This optimization problem will give values for $u$ at each time discretization node.
+
+To make this efficient, the optimization problem described above can be put into a function that uses $x_0$ as a parameter. In the code, this is used to return the optimal values of $x, u$. This can be done using the ``value`` function.
+
+For our purposes, we choose to run this program an arbitrary $400$ times, once every $0.05$ seconds.
+
+```julia
+x0 = [18 -14] # Initial point
+T = 400 # Length of simulation
+
+for t in T
+    x1, x2, u = MHMPC(x0)
+    x0 = [x1[2] x2[2]]
+end
+```
+
+Now, plotting the values of $x$ from our program:
```

Comment:
Can you add an extra empty line after line 206? In the markdown, this figure shows up on the same line as 206, and I think it would look a little better to have it below.